temporalio/worker/_activity.py

```diff
@@ -532,6 +532,19 @@ async def _run_activity(
         except Exception:
             temporalio.activity.logger.exception("Failed completing activity task")
 
+    def assert_activity_valid(self, activity) -> Optional[str]:
+        activity_def = self._activities.get(activity)
+        if self._dynamic_activity:
+            return None
```

Comment:
```suggestion
        if self._dynamic_activity:
            return None
        activity_def = self._activities.get(activity)
```

Pedantic, but can flip these